osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableHitStrong.cs

```diff
@@ -46,18 +44,26 @@ protected override void CheckJudgement(bool userTriggered)
                 Judgement.SecondHit = true;
         }
 
-        protected override bool HandleKeyPress(Key key)
+        public override bool OnReleased(TaikoAction action)
+        {
+            if (action == firstHitAction)
+                firstKeyHeld = false;

```

Comment:
Because the OnPressed event will arise once for each key, we only want to handle a successful strong hit if the circle was first hit with one key, and _held_ with that key while pressing the other.
So the way this is achieved is by storing the previous key pressed - the `firstKey`. Then `firstKeyHeld` is exactly what you have here - whether that `firstKey` was continually held down.